pkg/discovery/discovery.go

```diff
@@ -209,31 +231,31 @@ func (d *Discoverer) getMountOptionsFromStorageClass(name string) ([]string, err
 	return class.MountOptions, nil
 }
 
-func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConfig) {
+func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConfig) error {
 	klog.V(7).Infof("Discovering volumes at hostpath %q, mount path %q for storage class %q", config.HostDir, config.MountDir, class)
 
 	reclaimPolicy, err := d.getReclaimPolicyFromStorageClass(class)
 	if err != nil {
 		klog.Errorf("Failed to get ReclaimPolicy from storage class %q: %v", class, err)
-		return
+		return err
```

Comment:
I prefer the error handling style where the callers log the errors (to avoid double logging at every layer).

Reply:
Would you like me to move all the different logging lines in this function to the caller:
```go
(d *Discoverer) DiscoverLocalVolumes()
```
Don't you think we're going to loose some details about the error returned if we just log in the caller ?
It looks like all logs messages add more context to the returned error. Should we wrap them with this additional context ?

Reply:
How about `return fmt.Error("Failed to get ReclaimPolicy" ....)`

Reply:
I've done the changes, let me know if it matches what you proposed 👍